library/test/src/lib.rs

```diff
@@ -434,10 +439,12 @@ pub fn convert_benchmarks_to_tests(tests: Vec<TestDescAndFn>) -> Vec<TestDescAnd
         .map(|x| {
             let testfn = match x.testfn {
                 DynBenchFn(benchfn) => DynTestFn(Box::new(move || {
-                    bench::run_once(|b| __rust_begin_short_backtrace(|| benchfn(b)))
+                    bench::run_once(|b| __rust_begin_short_backtrace(|| benchfn(b)));
+                    Ok(())
```

Comment:
This looks off to me -- I would expect us to propagate the (possible) Result from benchfn through bench::run_once.

It looks like there's a similar problem in a few other places.

Reply:
Yeah, that's true -- a bench which returns Result::Err would be treated as having succeeded here. I've changed this to have bench::run_once also return a Result and adjusted the rest of the bench module accordingly.

I looked around to find a template from which to write a test to cover this, but couldn't find anything suitable. Any pointers would be welcome.

Reply:
I think in terms of testing probably the best bet is a UI test, similar to src/test/ui/rfc-1937-termination-trait/termination-trait-in-test.rs (that has a failing benchmark returning a Result, in fact, so maybe could've caught this? Not sure, I don't recall CI failing here, so perhaps not).

Reply:
I believe you have to use libtest directly, like for example compiletest does with DynamicTestFn.